### PR TITLE
Added parameter rollOnDateChange to AzureTableStorageSink

### PR DESCRIPTION
**Pull request by kashifsoofi**
Added parameter rollOnDateChange to AzureTableStorageSink to create a new table every day postfixed with Ticks for current date. We at WorldRemit need to pull a lot of data from log table to splunk and splunk at the moment cannot handle data once records exceed more than a million records hence need to break down the log data set.

### CODE REVIEW

**Comment by kashifsoofi** [comment]
Hi Nicholas,

Thanks for the quick review and a very positive and constructive feedback.
I do agree with the points raised.

Fair comment, I have updated the implementation to remove the specific
implementation of rollover and instead added ICloudTableProvider (similar
to formatProvider/keyGenerator). It is still very much an edge case but it
does provide the user the ability to provide their own implementation for
CloudTableProvider.

And of course having the sink added to our applications is an option but I
would rather be selfish and make contribution(s) upstream and keep
benefiting from the future improvements.

Kind Regards,
Kashif

**Comment by nblumhardt** [comment]
Thanks @kashifsoofi!

@serilog/reviewers-azure-table-storage and all, has anyone with more experience with this sink got feedback on the `ICloudTableProvider` approach? Sounds reasonable to me.

File: src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableProvider/DefaultCloudTableProvider.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+// limitations under the License.
+
+using System;
+using System.Threading;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Serilog.Sinks.AzureTableStorage.AzureTableProvider
+{
+    public class DefaultCloudTableProvider : ICloudTableProvider
```
Comment (nblumhardt) [inline src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableProvider/DefaultCloudTableProvider.cs]: Does this need to be public?

File: src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableProvider/DefaultCloudTableProvider.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        private readonly int _waitTimeoutMilliseconds = Timeout.Infinite;
```
Comment (nblumhardt) [inline src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableProvider/DefaultCloudTableProvider.cs]: Nit - the codebase doesn't use explicit `private` or `internal`

File: src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableProvider/DefaultCloudTableProvider.cs (hunk 1 above)
Comment (kashifsoofi) [reply src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableProvider/DefaultCloudTableProvider.cs]: Apologies for coming back late, yes it does not need to be, will update.

File: src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableProvider/DefaultCloudTableProvider.cs (hunk 2 above)
Comment (kashifsoofi) [reply src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableProvider/DefaultCloudTableProvider.cs]: Cool - will remove explicit uses of private and internal.

**Comment by nblumhardt** [comment]
Looks good!

Unfortunately I just realised that the PR is targeting the wrong branch (master rather than dev); switching it to `dev` causes a few conflicts because of code that has already moved forward on the current development version. I can circle back and help to resolve these, if it's not straightforward.

Sorry for not spotting this sooner.

**Comment by kashifsoofi** [comment]
I have merged the changes in dev branch and created a pull request #48, not sure why it started showing conflict here.

**Comment by nblumhardt** [comment]
In via #48

**Pull request merged**